### ISSUE TITLE
[KTIJ-29348] Rainbow Variables not working well on IntelliJ IDEA 2024.1 (Ultimate Edition) / Semantic highlighting is overridden

### ISSUE THREAD

**Issue by maor-graiber** [izhangzhihao/intellij-rainbow-brackets#2720]
### Have you checked the issues and discussions to ensure there are no duplicates?

Yes

### Your programming languages

Kotlin

### Free or paid?

Free users

### Expected Behavior

All my variables should have colors when "Rainbow Variables" are turned on.
See screenshots below:
![image](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/83915192/9fb3ac5a-b266-4ce1-bf2f-551259007e15)
![image](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/83915192/8cc8d27e-7c42-4d36-9cfd-4bd31a861b9f)

### Current Behavior

Only some variables have color while others don't.

### Code snippet for reproduce

```
class SomeClass {

  fun doSomething(firstInput: String, secondInput: Int) {

    // no color for concat variable (val)
    val concat = firstInput + secondInput
    println("Inside function block I see variable color but not in param list: $firstInput")
    println("Inside function block I see variable color but not in param list: $secondInput")

    println("here it has color: $concat")

    // no color for concat variable (var)
    var anotherOne = "asf"

    println("here it has color: $anotherOne")
  }

}
```

### Your Environment

IntelliJ IDEA 2024.1 (Ultimate Edition)
Build #IU-241.14494.240, built on March 28, 2024

Licensed to Next Software N.I.I ltd / Maor Graiber
Subscription is active until January 1, 2025.

Runtime version: 17.0.10+8-b1207.12 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.

**Comment by JakubKahovec** [comment]
Also happens for Python

**Comment by izhangzhihao** [comment]
This is a bug of the IDE, pls subscribe or vote for https://youtrack.jetbrains.com/issue/KTIJ-29348/Semantic-highlighting-is-overridden

**Comment by davichi11** [comment]
@izhangzhihao 所以这个要怎么解决呢？ 2024版是不是就没法用了？

**Comment by izhangzhihao** [comment]
this is an ide issue we can only wait for jetbrains to fix it.

**Comment by izhangzhihao** [comment]
The latest EAP build(2024.2 EAP) of the IntelliJ fixes this issue: 
<img width="1055" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/5cc5941c-4b22-4fee-9bc8-ea2e9b62eba9">

**Comment by izhangzhihao** [comment]
closed since it was fixed in 2024.2 EAP, pinned this issue for easy-to-search purposes.
